src/Compilers/CSharp/Portable/Symbols/Attributes/PEAttributeData.cs

```diff
@@ -102,7 +102,10 @@ private void EnsureAttributeArgumentsAreLoaded()
                 TypedConstant[]? lazyConstructorArguments = null;
                 KeyValuePair<string, TypedConstant>[]? lazyNamedArguments = null;
 
-                if (!_decoder.GetCustomAttribute(_handle, out lazyConstructorArguments, out lazyNamedArguments))
+                // If an attribute argument contains a generic parameter, the decoder needs to know the attribute type.
+                var decoderWithType = AttributeClass is { } c ? new MetadataDecoder(_decoder.ModuleSymbol, c) : _decoder;
```

Comment:
>var decoderWithType = AttributeClass is { } c ? new MetadataDecoder(_decoder.ModuleSymbol, c) : _decoder; [](http://example.com/codeflow?start=16&length=105)

It doesn't look like VB version of `PEAttributeData` was adjusted. Are we missing "interesting" VB tests?

Reply:
> Not sure if we are missing interesting VB tests 

I think we are, since none of the the tests caught the fact that we are not adjusting the context.

> at least the ones with generic attributes wouldn't work as generic attributes are not allowed in VB.

What exactly is not allowed:
- They are not allowed to be defined?
- They are not allowed to be applied?
- Generic attributes are not supported when found in metadata?

Shouldn't VB be able to import attributes emitted by C#? If it should, then all scenarios that we can test for C# we can test for VB.